hypothesis-python/src/hypothesis/_settings.py

```diff
@@ -702,55 +621,20 @@ def _validate_deadline(x):
 )
 
 
-class PrintSettings(Enum):
-    """Flags to determine whether or not to print a detailed example blob to
-    use with :func:`~hypothesis.reproduce_failure` for failing test cases."""
-
-    NEVER = 0
-    """Never print a blob."""
-
-    INFER = 1
-    """This option is deprecated and will be treated as equivalent to
-    ALWAYS."""
-
-    ALWAYS = 2
-    """Always print a blob on failure."""
-
-    def __repr__(self):
-        return "PrintSettings.%s" % (self.name,)
-
-
-def _validate_print_blob(value):
-    if isinstance(value, PrintSettings):
-        replacement = value != PrintSettings.NEVER
-
-        note_deprecation(
-            "Setting print_blob=%r is deprecated and will become an error "
-            "in a future version of Hypothesis. Use print_blob=%r instead."
-            % (value, replacement),
-            since="2018-09-30",
-        )
-        return replacement
-
-    check_type(bool, value, "print_blob")
-
-    return value
-
-
 settings._define_setting(
     "print_blob",
     default=False,
+    options=(True, False),
     description="""
 If set to True, Hypothesis will print code for failing examples that can be used with
 :func:`@reproduce_failure <hypothesis.reproduce_failure>` to reproduce the failing example.
 """,
-    validator=_validate_print_blob,
 )
 
 settings.lock_further_definitions()
 
 
-def note_deprecation(message, since, verbosity=None):
+def note_deprecation(message, since, verbosity=None):  # pragma: no cover
```

Comment:
I feel like we should just have explicit tests for `note_deprecation` because we're almost certainly going to need it again.

Reply:
#2285 has us covered for that, once I rebase everything will be covered.